python27Packages.faulthandler fails to build

## Issue description

python27Packages.faulthandler fails to build on the unstable channel

Cc @sauyon ([faulthandler's maintainer](https://github.com/NixOS/nixpkgs/blob/804285f589ded484b530750dd1ec03b9052bcdac/pkgs/development/python-modules/faulthandler/default.nix#L16))

### Steps to reproduce

```shell
 nix-channel --list
unstable https://nixos.org/channels/nixpkgs-unstable
```

```shell
nix-channel --update unstable
downloading Nix expressions from ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-18.03pre126020.931a0b8be80/nixexprs.tar.xz’...
downloading ‘https://d3g5gsiof5omrk.cloudfront.net/nixpkgs/nixpkgs-18.03pre126020.931a0b8be80/nixexprs.tar.xz’... [8271/9692 KiB, 813.3 KiB/s]
unpacking channels...
```

```shell
nix-env -iA unstable.python27Packages.faulthandler
```

The output shows test failures due to regular expressions not matching. E.g.

```
AssertionError: Regex didn't match: '^Fatal Python error: Segmentation fault\n\nCurrent\\ thread\\ XXX\\ \\<python\\>\\ \\(most\\ recent\\ call\\ first\\):\n  File "<string>", line 4 in <module>' not found in u'Fatal Python error: Segmentation fault\n\nCurrent thread XXX <python2.7> (most recent call first):\n  File "<string>", line 4 in <module>'
```
It seems that the tests are expecting to see `python` in the message, while the actual error says `python2.7`. If it's just that, I hope that it would be easy to fix the build.

I am happy to help if given some guidance.

Full output:
* https://gist.github.com/stefano-m/bf3f62b5597f79069adddf2d91b84c57

## Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 4.9.78, NixOS, 17.09.2875.c2b668ee726 (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-17.09.2875.c2b668ee726"`
 - channels(user): `"unstable-18.03pre126020.931a0b8be80"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

--- a/pkgs/development/python-modules/faulthandler/default.nix
+++ b/pkgs/development/python-modules/faulthandler/default.nix
@@ -1,7 +1,6 @@
-{ stdenv, fetchPypi, buildPythonPackage }:
+{ stdenv, fetchPypi, buildPythonPackage, fetchpatch }:
 
 buildPythonPackage rec {
-  name = "${pname}-${version}";
   pname = "faulthandler";
   version = "3.0";
 
@@ -9,6 +8,17 @@ buildPythonPackage rec {
     inherit pname version;
     sha256 = "acc10e10909f0f956ba1b42b6c450ea0bdaaa27b3942899f65931396cfcdd36a";
   };
+
+  patches = [
+    (fetchpatch {
+      url = https://github.com/vstinner/faulthandler/commit/67b661e.patch;
+      sha256 = "1nn8c9nq5qypja949hzz0n4yprsyr63wihf5g3gwrinm2nkjnnv7";
+    })
+    (fetchpatch {
+      url = https://github.com/vstinner/faulthandler/commit/07cbb7b.patch;
+      sha256 = "0fh6rjyjw7z1hsiy3sgdc8j9mncg1vlv3y0h4bplqyw18vq3srb3";
+    })
+  ];
 
   meta = {
     description = "Dump the Python traceback";

